public/r2/v1/swagger/swagger.json

```diff
@@ -184,6 +184,72 @@
                 }
             }
         },
+        "/namespaces/{namespaceID}/ruleset/bulk": {
+            "put": {
+                "tags":[
+                    "namespaces"
+                ],
+                "summary": "Bulk update endpoint for namespace changes",
+                "operationId": "bulkUpdateRuleSet",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "in": "path",
+                        "name": "namespaceID",
+                        "description": "The name of the namespace",
+                        "type": "string",
+                        "required": true
+                    },
+                    {
+                        "in": "body",
+                        "name": "ruleset-change",
+                        "description": "The updates to the ruleset",
+                        "required": true,
+                        "schema": {
+                        "$ref": "#/definitions/RuleSetChange"
+                        }
+                    },
+                    {
+                        "in": "body",
+                        "name": "rule-set-version",
+                        "description": "The ruleset version to update, for check and set operations",
```

Comment:
nit: might be more clear to say "The ruleset version to atomically apply bulk updates against"